File: 포키/20221018.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class BOJ11660 {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        long[][] map = new long[n+1][n+1];

        for (int i = 1; i <= n; i++) {
            StringTokenizer line = new StringTokenizer(br.readLine());
            for (int j = 1; j <= n; j++) {
                map[i][j] = map[i-1][j] + map[i][j-1] - map[i-1][j-1] + Long.parseLong(line.nextToken());
            }
        }

        for (int i = 0; i < m; i++) {
            StringTokenizer line = new StringTokenizer(br.readLine());
            int x1 = Integer.parseInt(line.nextToken());
            int y1 = Integer.parseInt(line.nextToken());
            int x2 = Integer.parseInt(line.nextToken());
            int y2 = Integer.parseInt(line.nextToken());

            long result = map[x2][y2] - map[x1 - 1][y2] - map[x2][y1 - 1] + map[x1 - 1][y1 - 1];

            System.out.println(result);
        }
    }

}

File: 후/20221018.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.lang.Integer.parseInt
import java.util.*

fun main() {
    val reader = BufferedReader(InputStreamReader(System.`in`))
    val writer = BufferedWriter(OutputStreamWriter(System.out))
    var tokenizer = StringTokenizer(reader.readLine())

    val N = parseInt(tokenizer.nextToken()) // 표의 크기
    val M = parseInt(tokenizer.nextToken()) // 합을 구해야 하는 횟수

    // init arr
    val arr = Array(N) { Array(N) { 0 } }
    val sumArr = Array(N) { Array(N) { 0 } }
    for (row in 0 until N) {
        var sum = 0
        tokenizer = StringTokenizer(reader.readLine())
        for (col in 0 until N) {
            arr[row][col] = parseInt(tokenizer.nextToken())
            sum += arr[row][col]
        }
    }

    for (i in 1..M) {
        tokenizer = StringTokenizer(reader.readLine())
        val x1 = parseInt(tokenizer.nextToken()) - 1
        val y1 = parseInt(tokenizer.nextToken()) - 1
        val x2 = parseInt(tokenizer.nextToken()) - 1
        val y2 = parseInt(tokenizer.nextToken()) - 1
        var sum = 0
        for (row in x1..x2) {
            for (col in y1..y2) {
                sum += arr[row][col]
            }
        }
        writer.append(sum.toString())
        writer.append(System.lineSeparator())
    }

    writer.close()
    reader.close()
}
